### PR TITLE
Backports for UPD-1009 / CP-50580

### PR DESCRIPTION
**Pull request by lindig**
* CA-395174: Try to unarchive VM's metrics when they aren..
* CA-395174: Try to unarchive VM's metrics when they aren..
* CA-394444: Update `vm_operation_table`                   
* CA-394444: Update task cancellation in `message_forward..
* CA-394169: Allow task to have permissions on itself - b..
* CA-394169: Allow task to have permissions on itself

### CODE REVIEW

**Comment by lindig** [comment]
https://github.com/xapi-project/xen-api-libs-transitional/pull/116 updates a library; the CI fails because it depends on it.

**Review by last-genius — APPROVED** [review]

File: ocaml/xapi/taskHelper.ml [hunk 1]
```diff
@@ -80,19 +89,14 @@ let assert_op_valid ?(ok_if_no_session_in_context = false) ~__context task_id =
...
-              ~self:task_session
-          in
-          let context_auth_user_sid =
-            Db_actions.DB_Action.Session.get_auth_user_sid ~__context
-              ~self:context_session
-          in
-          (*debug "task_auth_user_sid=%s,context_auth_user_sid=%s" task_auth_user_sid context_auth_user_sid;*)
-          task_auth_user_sid = context_auth_user_sid
+          (* If the task id is the same as the context's task id, we don't need
+             to go theough their respective sessions.*)
```
Comment (edwintorok) [inline ocaml/xapi/taskHelper.ml:93]: typo, but I assume it was in the original commit too.

**Comment by lindig** [comment]
Thanks for pointing out additional commits. I'll look into this. We really need to make sure commits have the correct CA as it's easy to miss related commits.

**Comment by lindig** [comment]
@edwintorok I've backported two more commits

**Comment by edwintorok** [comment]
DCO unhappy about a commit authored by @GabrielBuica with Co-authored-by Christian (and committer Christian), but I've set it to pass, we don't need to be so strict about this. To keep DCO happy it probably wants also a sign-off by Christian.

**Review by edwintorok — APPROVED** [review]

**Pull request merged**